Waiting for VPC Peering Connection to become available

I know that you're working on various VPC issues, so you may be aware of this problem.  I wanted to cover my bases and report it in case you're not.  

I'm using release 2.2.4 of moto.

I have two tests (written in Terraform) that I run against a Terraform module that sets up a peered connection.  It doesn't matter which order I run the tests, the second one fails with the error below.  I'm struggling to come up with self-contained tests to demonstrate the problem, but if the error message can't help point you at the problem, I'll work on the tests some more.

```
module.vpc_pcx_requester.aws_route_table_association.public[0]: Creation complete after 0s [id=rtbassoc-60df5245]
 
Error: Error waiting for VPC Peering Connection to become available: 
Error waiting for VPC Peering Connection (pcx-e88f13a3) to become available
unexpected state 'deleted', wanted target 'pending-acceptance, active'. last error: %!s(<nil>)

  with module.generated_route_tables.aws_vpc_peering_connection.this,
  on ../../main.tf line 15, in resource "aws_vpc_peering_connection" "this":
  15: resource "aws_vpc_peering_connection" "this" { !
make: *** [pytest//home/kbalk/plus3it/tardigrade-ci/tests/terraform_pytest] Error 1
```

--- a/moto/ec2/models.py
+++ b/moto/ec2/models.py
@@ -3563,7 +3563,9 @@ class VPCPeeringConnectionBackend(object):
                     vpc_pcx_cx.vpc_pcxs[vpc_pcx_id] = vpc_pcx
         return vpc_pcx
 
-    def get_all_vpc_peering_connections(self):
+    def get_all_vpc_peering_connections(self, vpc_peering_ids=None):
+        if vpc_peering_ids:
+            return [pcx for pcx in self.vpc_pcxs.values() if pcx.id in vpc_peering_ids]
         return self.vpc_pcxs.values()
 
     def get_vpc_peering_connection(self, vpc_pcx_id):

--- a/moto/ec2/responses/vpc_peering_connections.py
+++ b/moto/ec2/responses/vpc_peering_connections.py
@@ -29,7 +29,8 @@ class VPCPeeringConnections(BaseResponse):
         return template.render(vpc_pcx=vpc_pcx)
 
     def describe_vpc_peering_connections(self):
-        vpc_pcxs = self.ec2_backend.get_all_vpc_peering_connections()
+        ids = self._get_multi_param("VpcPeeringConnectionId")
+        vpc_pcxs = self.ec2_backend.get_all_vpc_peering_connections(vpc_peering_ids=ids)
         template = self.response_template(DESCRIBE_VPC_PEERING_CONNECTIONS_RESPONSE)
         return template.render(vpc_pcxs=vpc_pcxs)
 

--- a/tests/test_ec2/test_vpc_peering.py
+++ b/tests/test_ec2/test_vpc_peering.py
@@ -228,6 +228,36 @@ def test_vpc_peering_connections_cross_region_fail():
 
 
 @mock_ec2
+def test_describe_vpc_peering_connections_only_returns_requested_id():
+    # create vpc in us-west-1 and ap-northeast-1
+    ec2_usw1 = boto3.resource("ec2", region_name="us-west-1")
+    vpc_usw1 = ec2_usw1.create_vpc(CidrBlock="10.90.0.0/16")
+    ec2_apn1 = boto3.resource("ec2", region_name="ap-northeast-1")
+    vpc_apn1 = ec2_apn1.create_vpc(CidrBlock="10.20.0.0/16")
+    # create peering
+    vpc_pcx_usw1 = ec2_usw1.create_vpc_peering_connection(
+        VpcId=vpc_usw1.id, PeerVpcId=vpc_apn1.id, PeerRegion="ap-northeast-1"
+    )
+    vpc_pcx_usw2 = ec2_usw1.create_vpc_peering_connection(
+        VpcId=vpc_usw1.id, PeerVpcId=vpc_apn1.id, PeerRegion="ap-northeast-1"
+    )
+    # describe peering
+    ec2_usw1 = boto3.client("ec2", region_name="us-west-1")
+    all_pcx = ec2_usw1.describe_vpc_peering_connections()["VpcPeeringConnections"]
+    all_pcx.should.have.length_of(2)
+
+    both_pcx = ec2_usw1.describe_vpc_peering_connections(
+        VpcPeeringConnectionIds=[vpc_pcx_usw1.id, vpc_pcx_usw2.id]
+    )["VpcPeeringConnections"]
+    both_pcx.should.have.length_of(2)
+
+    one_pcx = ec2_usw1.describe_vpc_peering_connections(
+        VpcPeeringConnectionIds=[vpc_pcx_usw1.id]
+    )["VpcPeeringConnections"]
+    one_pcx.should.have.length_of(1)
+
+
+@mock_ec2
 def test_vpc_peering_connections_cross_region_accept():
     # create vpc in us-west-1 and ap-northeast-1
     ec2_usw1 = boto3.resource("ec2", region_name="us-west-1")

